src/frontend/web_application/src/scenes/Discussion/components/MessageAttachments/index.jsx

```diff
@@ -13,8 +16,104 @@ class MessageAttachments extends Component {
   static defaultProps = {
   };
 
+  state = {
+    downloads: {},
+  };
+
+  // eslint-disable-next-line react/sort-comp
+  downloadRefs = {};
+
+  cancelTokenSource = CancelToken.source();
+
+  componentWillUnmount() {
+    this.cancelTokenSource.cancel();
+  }
+
+  getRef = (index) => {
+    if (!this.downloadRefs[index] || !this.downloadRefs[index].ref) {
+      this.downloadRefs[index] = {
+        ...this.downloadRefs[index],
+        ref: createRef(),
+      };
+    }
+
+    return this.downloadRefs[index].ref;
+  }
+
+  sendFileToBrowser = (index, { data, filename, mime }) => {
+    const blob = new Blob([data], { type: mime || 'application/octet-stream' });
+    const blobURL = window.URL.createObjectURL(blob);
+    const link = document.createElement('a');
+    link.style.display = 'none';
+    link.href = blobURL;
+    link.setAttribute('download', filename);
+    this.downloadRefs[index].ref.current.appendChild(link);
+    link.click();
```

Comment:
I'm clearly not a fan of this kind of hack… No other way to make DL happen ?

Reply:
I didn't find other way to send the file to the browser. cf. https://github.com/kennethjiang/js-file-download/blob/master/file-download.js

Reply:
… or may be https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/downloads/download
I'll check that

Reply:
arf no, it's an API for WebExtensions